pdk.gemspec

```diff
@@ -3,25 +3,26 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'pdk/version'
 
 Gem::Specification.new do |spec|
-  spec.name    = 'pdk'
+  spec.name = 'pdk'
   spec.version = PDK::VERSION
   spec.authors = ['Puppet, Inc.']
-  spec.email   = ['pdk-maintainers@puppet.com']
+  spec.email = ['pdk-maintainers@puppet.com']
 
-  spec.summary     = 'A key part of the Puppet Development Kit, the shortest path to better modules'
+  spec.summary = 'A key part of the Puppet Development Kit, the shortest path to better modules'
   spec.description = 'A CLI to facilitate easy, unified development workflows for Puppet modules.'
-  spec.homepage    = 'https://github.com/puppetlabs/pdk'
+  spec.homepage = 'https://github.com/puppetlabs/pdk'
 
-  spec.files         = Dir['CHANGELOG.md', 'README.md', 'LICENSE', 'lib/**/*', 'exe/**/*', 'locales/**/*']
-  spec.bindir        = 'exe'
-  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.files = Dir['CHANGELOG.md', 'README.md', 'LICENSE', 'lib/**/*', 'exe/**/*', 'locales/**/*']
+  spec.bindir = 'exe'
+  spec.executables = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
   spec.required_ruby_version = '>= 2.1.9'
 
+  spec.add_runtime_dependency 'addressable', '2.6.0'
   spec.add_runtime_dependency 'bundler', '>= 1.15.0', '< 3.0.0'
   spec.add_runtime_dependency 'childprocess', '~> 0.7.1'
-  spec.add_runtime_dependency 'cri', '~> 2.10'
+  spec.add_runtime_dependency 'cri', '= 2.10.1'
```

Comment:
This makes PDK bundle incompatible with `r10k` which was an issue for people previously. (See also: https://github.com/puppetlabs/pdk/pull/731)

Reply:
This is required for 2.1 support unless we do a <= 2.10.1

Reply:
It's not _required_ since `2.10.1` satisfies `~> 2.10`, it just means you have to utilize some other means to pin `cri` since Rubygems is so dumb about how it considers Ruby version requirements in gems. I think it would be better for us to document those other means than break the co-installability of `pdk` with a growing number of other ecosystem tools for the people on modern Rubies...

Reply:
`~> 2.10` would pull in 2.15 which starts to bring in dependencies that have requirement issues with ruby 2.1

Reply:
It should only pull in the latest if it's not already satisfied by an existing version of the gem.